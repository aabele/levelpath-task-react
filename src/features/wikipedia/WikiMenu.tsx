import React from 'react';
import { Menu, Layout } from 'antd';

interface Props {
  items: string[];
  activeKey: string;
  setActive(item: string): void;
}

const capFirst = (data: string) => data.charAt(0).toUpperCase() + data.slice(1);

export const WikiMenu: React.FC<Props> = ({
  items,
  activeKey,
  setActive,
}) => (
  <Layout.Header>
    <Menu
      theme="dark"
      mode="horizontal"
      activeKey={activeKey}
      selectedKeys={[activeKey]}
      onClick={(item) => setActive(item.key)}
      items={ items.map((item: string) => ({ key: item, label: capFirst(item)}))}
    />
  </Layout.Header>
)
