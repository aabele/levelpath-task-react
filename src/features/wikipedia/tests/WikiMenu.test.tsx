import React from 'react';
import { render, fireEvent, act, waitFor, screen } from '@testing-library/react';
import { WikiMenu } from '../WikiMenu';

describe('WikiMenu', () => {
  it('should render properly', async () => {
    const items = ['item1', 'item2']
    await act(async () => {
      const { baseElement } = await render(<WikiMenu items={items} activeKey={items[0]} setActive={() => {}} />);
      await expect(baseElement).toMatchSnapshot();
    });
  });

  it('should trigger callback on item click', async () => {
    const items = ['Item1', 'Item2']
    const setActiveKey = jest.fn();
    await act(async () => {
      await render(<WikiMenu items={items} activeKey={items[0]} setActive={setActiveKey} />);
    })
    const item = await waitFor(() => screen.getByText(/Item1/));
    fireEvent.click(item);
    await expect(setActiveKey).toHaveBeenCalledTimes(1);
  });

});
