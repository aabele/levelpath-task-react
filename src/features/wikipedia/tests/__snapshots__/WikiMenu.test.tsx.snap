// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WikiMenu should render properly 1`] = `
<body>
  <div />
</body>
`;
